From 7c29999bfd7b4b24e01d3e432133a3d4438ab15e Mon Sep 17 00:00:00 2001
From: Dong Aisheng <b29396@freescale.com>
Date: Thu, 30 Apr 2015 18:32:30 +0800
Subject: [PATCH] MLK-10853-1 clk-imx6ul: set CAN_SEL parent to CLK_PLL3_60M

The default parent is dummy clock which can not work for CAN devices.
Change the parent to CLK_PLL3_60M.
Due to it's not board dependant, so we make the change in SoC clock driver.

Signed-off-by: Dong Aisheng <b29396@freescale.com>
---
 arch/arm/mach-imx/clk-imx6ul.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/arch/arm/mach-imx/clk-imx6ul.c b/arch/arm/mach-imx/clk-imx6ul.c
index 0050b65..21511cc 100644
--- a/arch/arm/mach-imx/clk-imx6ul.c
+++ b/arch/arm/mach-imx/clk-imx6ul.c
@@ -411,6 +411,8 @@ static void __init imx6ul_clocks_init(struct device_node *ccm_node)
 		imx_clk_prepare_enable(clks[IMX6UL_CLK_USBPHY2_GATE]);
 	}
 
+	imx_clk_set_parent(clks[IMX6UL_CLK_CAN_SEL], clks[IMX6UL_CLK_PLL3_60M]);
+
 	/* Set initial power mode */
 	imx6q_set_lpm(WAIT_CLOCKED);
 
-- 
1.8.0

